pinot-core/src/main/java/com/linkedin/pinot/core/operator/query/DictionaryBasedAggregationOperator.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.operator.query;
+
+import com.linkedin.pinot.common.segment.SegmentMetadata;
+
+import com.linkedin.pinot.core.operator.BaseOperator;
+import com.linkedin.pinot.core.operator.ExecutionStatistics;
+import com.linkedin.pinot.core.operator.blocks.IntermediateResultsBlock;
+import com.linkedin.pinot.core.query.aggregation.AggregationFunctionContext;
+import com.linkedin.pinot.core.query.aggregation.AggregationResultHolder;
+import com.linkedin.pinot.core.query.aggregation.DoubleAggregationResultHolder;
+import com.linkedin.pinot.core.query.aggregation.ObjectAggregationResultHolder;
+import com.linkedin.pinot.core.query.aggregation.function.AggregationFunction;
+import com.linkedin.pinot.core.query.aggregation.function.AggregationFunctionFactory;
+import com.linkedin.pinot.core.query.aggregation.function.customobject.MinMaxRangePair;
+import com.linkedin.pinot.core.segment.index.readers.Dictionary;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Aggregation operator that utilizes metadata for serving aggregation queries.
+ */
+public class DictionaryBasedAggregationOperator extends BaseOperator<IntermediateResultsBlock> {
+  private static final String OPERATOR_NAME = "DictionaryBasedAggregationOperator";
+
+  private final AggregationFunctionContext[] _aggregationFunctionContexts;
+  private final Map<String, Dictionary> _dictionaryMap;
+  private final SegmentMetadata _segmentMetadata;
+  private ExecutionStatistics _executionStatistics;
+
+  /**
+   * Constructor for the class.
+   *
+   * @param aggregationFunctionContexts Aggregation function contexts.
+   * @param segmentMetadata Segment metadata.
+   * @param dictionaryMap Map of column to its dictionary.
+   */
+  public DictionaryBasedAggregationOperator(AggregationFunctionContext[] aggregationFunctionContexts,
```

Comment:
It seems the approach here was to not use MetadataBasedOperator, and if so, then we should not need metadata here. If the only need for that is to get totalRawDocs, may be we should just pass that?